### PR TITLE
Clarify Strong Parameters documentation

### PR DESCRIPTION
**Pull request by keithpitty**
### Summary

The Rails Guides contain some misleading statements with regard to the way strong parameters are handled in the case where a required parameter is missing. This has [caused confusion](https://github.com/rails/rails/issues/27128).

This change aims to clarify the Rails Guides, specifically the _Action Controller Overview_ and _Configuring Rails Application_ guides. 

Hopefully, as a result, Rails users will understand that the `ActionController::MissingParameter` exception is not actually rescued so that a 400 Bad Request response can be sent. The `rescue_responses` hash is actually used by `ActionDispatch::ExceptionWrapper` to map exceptions to status codes for `ActionController::LogSubscriber`. As a result the mapped status codes appear in the log.

### CODE REVIEW

File: guides/source/configuring.md [hunk 1]
```diff
@@ -465,7 +465,7 @@ encrypted cookies salt value. Defaults to `'signed encrypted cookie'`.
   method should be performed on the parameters. See [Security Guide](security.html#unsafe-query-generation)
   for more information. It defaults to `true`.
 
-* `config.action_dispatch.rescue_responses` configures what exceptions are assigned to an HTTP status. It accepts a hash and you can specify pairs of exception/status. By default, this is defined as:
+* `config.action_dispatch.rescue_responses` configures what exceptions are assigned to an HTTP status. It accepts a hash and you can specify pairs of exception/status. Please note that this is only for the purposes of logging. No exceptions are actually rescued. By default, this is defined as:
```
Comment (matthewd) [inline guides/source/configuring.md]: That's rather a matter of perspective, I think...

The exception will still lead to an error response, but it is rescued from being a generic 500, and instead processed by a handler that will return a more pertinent HTTP status.

**Review by matthewd — COMMENTED** [review]
👋🏻!

I think the point at issue in #27128 is just the false claim about the default value... while the large hash shown is the *effective* default after the config is loaded, most of them never actually appear in the config hash itself.

**Comment by matthewd** [comment]
> Rails users will understand that the ActionController::MissingParameter exception is not actually rescued so that a 400 Bad Request response can be sent

Sorry, missed this part: yes, it is. If you're not actually getting a 400 (.. at least in production?), something is broken.

File: guides/source/configuring.md (hunk 1 above)
Comment (keithpitty) [inline guides/source/configuring.md]: Apologies, my mistake. I made some wrong assumptions that led me up the garden path! I blame the man flu from which I have been recovering.

I now see that the exception is rescued from within `ActionPack::ShowExceptions`.

**Comment by keithpitty** [comment]
Closing. This PR was raised due to my misunderstanding.

**Comment by keithpitty** [comment]
@matthewd thanks for setting me straight.

**Pull request closed without merging**